satellite/admin-back-office-v1.md

```diff
@@ -0,0 +1,554 @@
+authentication proxy requires to establish links between the networks, so the proxy can access to
+all the services. A solution for not having to peer the networks is to have an authentication proxy
+for every satellite, which also mitigates the single point of failure of a single authentication
+proxy.
+
+When using an authentication proxy for each service we have to configure them in a way that users
+don't have to log in more than once when they try to access to a different satellite that the one
+that they logged in for satisfying the _users stories_.
+
+For Storj DCS, we determined that the we want to authenticate the users through Oauth2 with their
+Storj email account, which is hosted by Google Workspace. We are going to authorize assigning them
+to different groups and mapping those groups to the back office roles.
+
+For authenticating the users we will use one [Oauth2-proxy](https://oauth2-proxy.github.io/oauth2-proxy/)
+for each satellite and for the "service" that will show the list of satellites, henceforth "selector
+service".
+
+All the Oauth2-proxies must use the same Oauth2 client credentials for not requiring the users to
+log in when they access to a different satellite or from the "selector service" once they logged in
+successfully in one of them and all the satellites and the "selector service" must be hosted under
+the same base domain or sub-domain. You can see [this proof of concept](https://github.com/ifraixedes/poc-oauth2-proxies-single-auth)
+for how to achieve it.
+
+##### API implementation
+
+We have to review the current satellite administration API to see what endpoints we need to add or
+modify.
+
+We also have to adapt it to allow to perform the operations based on the usr roles specified in the
+_Authentication / Authorization_ section and be able to configure the groups associated to them.
+
+We have to review it to see if we want to add more logging or metrics, considering that wasn't
+created as a main feature as now.
+
+##### Web UI
+
+We have to build a new Web UI from scratch with the Vue framework and the Vuetify component
+framework. The Web UI is already in development and it almost have every component and page that the
+back office requires, you can see it on https://storj.github.io/admin-ui/login.
+
+We need to add the API calls and UI logic (client side validations, operation confirmation, etc.)
+and all the Javascript client side code that those require.
+
+The current admin web UI will be deleted once this new one is rolled on.
+
+## Reminders
+
+### Security
+
+Everything related to uses' authentication relies on the external authentication proxy. A shallow
+list of proxy responsibilities are:
+- Users account management.
+- Users' sessions management.
+- Security on the authentication method.
+
+Because of the operations that the back office allows, we have to pay special attention and take all
+the measures that we can for protecting the site from Cross-Site Request Forgery (a.k.a. CSRF).
```

Comment:
There's also https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html

Reply:
I mentioned those cheatsheets series below (last paragraph of this section).
However, I didn't find an index page, so I linked the first cheatsheet of the series. 
I'm going to update the link